packages/solid/src/render/flow.ts

```diff
@@ -258,3 +259,23 @@ export function ErrorBoundary(props: {
     "_SOLID_DEV_" ? { name: "value" } : undefined
   ) as Accessor<JSX.Element>;
 }
+
+export function ClientOnly(props: {
+  children?: JSX.Element,
+}): JSX.Element {
+  if (sharedConfig.context) {
```

Comment:
iirc this is the way to check if the app is currently hydrating. This check is needed so that UI correction can be applied, otherwise we can just render the children directly (since the app is just in CSR)